src/CMakeLists.txt

```diff
@@ -170,6 +170,10 @@ macro (add_compiler_flag flag)
   set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} ${flag}")
 endmacro (add_compiler_flag)
 
+# Disallow undefined symbols in object files. Undefined symbols in shared
+# libraries are still allowed.
```

Comment:
I copied this from ld's man page. According to http://stackoverflow.com/questions/1617286/easy-check-for-unresolved-symbols-in-shared-libraries, passing '-Wl,-z,defs' will just force us to specify all dependent libraries on every library's link line, whereas not passing it just leaves the symbols undefined temporarily and those symbols, of course, will have to be present at the final link.